70_FLIPFluids/main.cpp

```diff
@@ -23,2218 +18,1845 @@ using namespace asset;
 using namespace ui;
 using namespace video;
 
+#include "app_resources/common.hlsl"
+#include "app_resources/gridUtils.hlsl"
+#include "app_resources/render_common.hlsl"
+#include "app_resources/descriptor_bindings.hlsl"
 
 enum SimPresets
 {
-	CENTER_DROP,
-	LONG_BOX
-};
-
-struct Particle
-{
-	float32_t4 position;
-    float32_t4 velocity;
-
-    uint32_t id;
-    uint32_t pad[3];
-};
-
-struct SGridData
-{
-    float gridCellSize;
-    float gridInvCellSize;
-    float pad0[2];
-
-    int32_t4 particleInitMin;
-    int32_t4 particleInitMax;
-    int32_t4 particleInitSize;
-
-    float32_t4 worldMin;
-    float32_t4 worldMax;
-    int32_t4 gridSize;
+    CENTER_DROP,
+    LONG_BOX
 };
 
 struct SMVPParams
 {
-	float cameraPosition[4];
+    float cameraPosition[4];
```

Comment:
the MVP stuff is still outstanding to share and use `nbl::hlsl::` types for

Reply:
I think you only need MVP, one row of V (the worldspace of the viewspace X+ direction), and not the M or the P on their own

Enough to fit into push constants